### PR TITLE
[WIP] JSTOR Pdf proxing

### PR DESCRIPTION
**Pull request by marcospri**
```
export JSTOR_SECRET_IP=3.141.59.26
export JSTOR_URL=https://b49tnk5c88.execute-api.us-east-1.amazonaws.com/prod/pdf
```
- make dev

- `http://0.0.0.0:9082/pdf?url=https://www.jstor.org/stable/1234`

### CODE REVIEW

File: tests/unit/services.py [hunk 1]
```diff
@@ -45,3 +46,11 @@ def http_service(mock_service):
 @pytest.fixture
 def pdf_url_builder_service(mock_service):
     return mock_service(PDFURLBuilder)
+
+
+@pytest.fixture
```
Comment (marcospri) [inline tests/unit/services.py:51]: Just enough testing changes to get the current ones to pass.

File: tox.ini [hunk 2]
```diff
@@ -48,6 +48,8 @@ passenv =
     dev: NGINX_SERVER
     dev: CLIENT_EMBED_URL
     dev: SIGNED_URLS_REQUIRED
+    dev: JSTOR_SECRET_IP
+    dev: JSTOR_URL
```
Comment (marcospri) [inline tox.ini:52]: Maybe JSTOR_API_URL?

File: via/views/view_pdf.py [hunk 3]
```diff
@@ -67,6 +67,14 @@ def proxy_google_drive_file(request):
     return _iter_pdf_response(request.response, content_iterable)
 
 
+@view_config(route_name="proxy_jstor_pdf", decorator=(has_secure_url_token,))
+def proxy_jstor_pdf(request):
+    content_iterable = request.find_service(JSTORAPI).jstor_pdf_stream(
+        request.params["doi"]
```
Comment (marcospri) [inline via/views/view_pdf.py:73]: I think we should pass this as part of path as other pdf enpoints in via do. Then the only query param will be the original URL.

File: via/services/jstor.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        self._http = http_service
+
+    def doi_from_url(self, url: str) -> Optional[str]:
+        parsed = urlparse(url)
+        url = parsed._replace(query="", fragment="").geturl()
+
+        for regexp in JSTORAPI.DOI_REGEXP:
+            if match := regexp.match(url):
+                if not match.groupdict().get("prefix"):
+                    return f"{JSTORAPI.DEFAULT_DOI_PREFIX}/{match['doi']}"
```
Comment (marcospri) [inline via/services/jstor.py:30]: Only tested this codepath.

File: tox.ini (hunk 2 above)
Comment (robertknight) [reply tox.ini:52]: `JSTOR_PDF_URL`?

**Comment by marcospri** [comment]
Closed in favour of https://github.com/hypothesis/via/pull/701

**Pull request closed without merging**